Code address broken setter/getter functions found in the surface bmi

rmcd-mscb opened the pull request:
@mdpiper, I looped through all output vars and queried as the following functions below and systematically went through the Fortran code to fix.  I hope I got it all.  I did the same getvarinfo with setter vars and fixed missing type, units, size, but have still not testing setting values to every var: 
``` python
def getvarinfo(var):
    # Get variable info.
    var_name = var   # vector
    # var_name = 'nhru'      # scalar
    # var_name = 'dday_slope'  # rectilinear
    print('Variable {}'.format(var_name))
    print(' - variable type:', m.var_type(var_name))
    print(' - units:', m.var_units(var_name))
    print(' - itemsize:', m.var_itemsize(var_name))
    print(' - nbytes:', m.var_nbytes(var_name))
    print(' - location:', m.var_location(var_name))
def getdata(var):
    try:
        print(m.get_value(var), flush=True)
    except:
        print(f'error with {var}', flush=True)
```

mdpiper commented on examples/testing_helpers.f90:
```diff
@@ -1,7 +1,7 @@
 
     ! character (len=*), parameter :: config_file = "control.simple1x"
-    character (len=*), parameter :: config_file = "control.simple1y"
+    character (len=*), parameter :: config_file = "pipestem/control.simple1"
```
I can't get this to work. I've only been able to run PRMS if the control file is in $PWD. Here, I get:
```
$ ls pipestem/
control.default  daymet.nc        model.out        snowcomp.wbal
control.simple1  gmon.out         myparam.nc       soilzone.wbal
control.simple1x gwflow.wbal      output           waterbal.wbal
control.simple1y intcp.wbal       sf_data          wbal.msgs

$ prms pipestem/control.simple1
 CLOSEZERO, NEARZERO, DNEARZERO
   1.19209290E-07   1.19209290E-07   2.2204460492503131E-016
 Ranges
 r32:           37
 smallest r32 value:    1.17549435E-38
 minexponent of r32:         -125
 r64:            2       -1021        1024
 smallest r64 value:    2.2250738585072014E-308
 precision of r64:           15
 range of r64:          307
========================================================================
Control File                   Control_class        2019-02-01 13:50:00Z
 No such file or directory
STOP Stopped
```

rmcd-mscb replied:
@mdpiper Ok - no problem I can work with the config_file without pipestem/.

mdpiper replied:
I still think this should be changed. If you'd prefer to use the path to the control file, I think (it's my understanding) that the control file needs paths, as well. Currently, `irf_ex` passes because PRMS exits 0 (success) even though it can't locate files referenced in the control file (this is my interpretation of the error message above without digging too far into the PRMS code), which signals success to `ctest`.

mdpiper commented:
@rmcd-mscb Just to be careful, you don't have to do anything with the tests in the `tests` directory--those are unit tests, used to make sure the BMI functions have the right names and arguments, and that they return a simple expected result. What I would really appreciate your help with is the tests in the `examples` directory (which I didn't name very well)--these are system tests, making sure the model, through its BMI, is behaving as expected. I don't know PRMS like you do, so I can't do a good job with these.

rmcd-mscb commented:
Thanks @mdpiper.  Regarding the unit tests I wanted to run a unit test on setting tmin, tmax, and prcp, since I changed the behavior in PRMS6,.  So far the irf_ex are pretty good system tests, AND I'll think of some more ways to push it.  I think the Travis fails are because I changed PRMS.  I was waiting to comment until those finished hoping they would be successful but again I think once PRMS is rebuilt these will pass (hopefully), if not I'll take care of it.  The surface BMI should be ready to go!

mdpiper commented:
@rmcd-mscb I updated PRMS to the latest from yesterday & ran the tests here again, but they're failing.

rmcd-mscb commented:
Mark - I don't want to send you down the wrong turn but did you update prms from my fork of the 6.0.0_dev_bmi branch?  

FOr example those tmax_in, and tmin_in are in teh interface: https://github.com/rmcd-mscb/prms/blob/a2aea90675e89103d87c18d5de7b07f11fa105b7/src/prmslib/physics/c_temperature.f90#L72

if you did update from my branch then I'll double check and get back to you

mdpiper commented:
@rmcd-mscb I built PRMS on https://github.com/rmcd-mscb/prms/commit/a2aea90675e89103d87c18d5de7b07f11fa105b7.

rmcd-mscb commented:
Hi @mdpiper - sorry I completely spazzed yesterday.  I apologize!!  I think I've got this fixed.  I found a solution for the compile fail here: https://philbull.wordpress.com/2012/09/07/weird-gfortran-compiler-error-of-the-week/ which is partly why I spazzed because it was compiling on my end, and I had looked at gfortran docs specifically when I made that change to be sure I wouldn't get an intel/gfortran mis-match.

Also - we've talked about this before but I can't remember the result.  Some vars like tmax, tmin, hru_ppt would be dangerous to set by indices so they have only value set_value function - is that all right?  At some point I'll have to create documentation and I can spell that out.

A couple of the simple unit tests failed and I fixed those as well :)

Mark - I noticed that I was running the temperature module in the soil-bmi which I don't need to do since its run in the surface, and now that I added setter functions for temp to the surface-bmi this would defeat the purpose.  So I'm going to push another fix to the soil-bmi.  I'm sorry sometimes there is a lot to keep track of here.  So you know I reran my notebook comparisons of the output to make sure everything is working as expected and with these changes I'm still getting the same output from the surface-soil coupling as I am with prms6 itself.  Thanks for your patience Mark - I really appreciate it!

Finally - really confused about Travis perhaps we can revisit that some time.

mdpiper commented on .travis.yml:
```diff
@@ -32,6 +32,8 @@ install:
   - mkdir _build && cd _build
   - cmake .. -DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX
   - make
+  - make clean
+  - make
```
Travis starts with a clean env, so this doesn't do anything. It should be removed.

rmcd-mscb replied:
Ok - but this was the **fix** that got past the initial fail to compile in Travis.  Which was challenging because it did compile and run for me with my intel setup.  It was the only thing I could find to get past it.

rmcd-mscb commented:
Thanks for your patience on this one Mark!
